Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.cpp

```diff
@@ -114,50 +114,64 @@ void RemoteMediaRecorderPrivateWriterManager::create(RemoteMediaRecorderPrivateW
 {
     ASSERT(!m_remoteMediaRecorderPrivateWriters.contains(identifier));
 
-    m_remoteMediaRecorderPrivateWriters.add(identifier, RemoteMediaRecorderPrivateWriterProxy::create(mimeType));
+    m_remoteMediaRecorderPrivateWriters.add(identifier, Writer { RemoteMediaRecorderPrivateWriterProxy::create(mimeType) });
 }
 
 void RemoteMediaRecorderPrivateWriterManager::addAudioTrack(RemoteMediaRecorderPrivateWriterIdentifier identifier, RemoteAudioInfo info, CompletionHandler<void(std::optional<uint8_t>)>&& completionHandler)
 {
     ASSERT(m_remoteMediaRecorderPrivateWriters.contains(identifier));
+    auto iterator = m_remoteMediaRecorderPrivateWriters.find(identifier);
+    if (iterator == m_remoteMediaRecorderPrivateWriters.end())
+        return;
 
-    m_audioInfo = info.toAudioInfo();
-    RefPtr writer = m_remoteMediaRecorderPrivateWriters.get(identifier);
-    auto result = writer->addAudioTrack(Ref { *m_audioInfo });
+    Ref audioInfo = info.toAudioInfo();
+    iterator->value.audioInfo = audioInfo.ptr();
+    RefPtr writer = iterator->value.proxy;
+    auto result = writer->addAudioTrack(audioInfo.get());
     if (result)
-        m_audioInfo->trackID = *result;
+        audioInfo->trackID = *result;
     completionHandler(result);
 }
 
 void RemoteMediaRecorderPrivateWriterManager::addVideoTrack(RemoteMediaRecorderPrivateWriterIdentifier identifier, RemoteVideoInfo info, std::optional<CGAffineTransform> transform, CompletionHandler<void(std::optional<uint8_t>)>&& completionHandler)
 {
     ASSERT(m_remoteMediaRecorderPrivateWriters.contains(identifier));
+    auto iterator = m_remoteMediaRecorderPrivateWriters.find(identifier);
+    if (iterator == m_remoteMediaRecorderPrivateWriters.end())
+        return;
 
-    m_videoInfo = info.toVideoInfo();
-    RefPtr writer = m_remoteMediaRecorderPrivateWriters.get(identifier);
-    auto result = writer->addVideoTrack(Ref { *m_videoInfo }, transform);
+    Ref videoInfo = info.toVideoInfo();
+    iterator->value.videoInfo = videoInfo.ptr();
+    RefPtr writer = iterator->value.proxy;
+    auto result = writer->addVideoTrack(videoInfo.get(), transform);
     if (result)
-        m_videoInfo->trackID = *result;
+        videoInfo->trackID = *result;
     completionHandler(result);
 }
 
 void RemoteMediaRecorderPrivateWriterManager::allTracksAdded(RemoteMediaRecorderPrivateWriterIdentifier identifier, CompletionHandler<void(bool)>&& completionHandler)
 {
     ASSERT(m_remoteMediaRecorderPrivateWriters.contains(identifier));
 
```

Comment:
I'd prefer if it used the same pattern as elsewhere:
```
    if (iterator == m_remoteMediaRecorderPrivateWriters.end())
        return;
```

and use iterator otherwise. 

`RefPtr writer = iterator->value.proxy;`